main.cpp

```diff
@@ -101,17 +91,37 @@ int main()
      construct 2 more Car instances via emplace_back.
      */
     
+    cars.emplace_back("allen");
+    cars.emplace_back("betty");
+    
     /*
      now reserve and emplace_back your Trucks and Motorcyles
      */
+    motorcycles.reserve(1);
+    motorcycles.emplace_back("charles");
     
+    trucks.reserve(1);
+    trucks.emplace_back("dave");
     
-    
-    
-    assert(false);
+    //assert(false);
     //add the cars, motorcycles and trucks to the highway using range-based for() loops: for( element : vec ) { ... }
     //be careful to not accidentally make element copies when iterating.
     
+    for (Vehicle& element : cars)
+    {
+        highway.addVehicle(&element);
+    }
+    
+    for (Vehicle& element : motorcycles)
+    {
+        highway.addVehicle(&element);
+    }
+    
+    for (Vehicle& element : trucks)
+    {
+        highway.addVehicle(&element);
+    }
```

Comment:
by not using `auto&` you are casting your vector elements to `Vehicle&`, instead of whatever type they are.
- [ ] replace `Vehicle&` with `auto&`